[CI] GenerativeIT test failing

**Build Scans:**
- [elasticsearch-periodic #12001 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/3grxfpiwyp6bo)
- [elasticsearch-pull-request #128476 / part-3](https://gradle-enterprise.elastic.co/s/5nlxpq5l7wl3w)
- [elasticsearch-pull-request #128424 / part-3](https://gradle-enterprise.elastic.co/s/hjhrstcsiuq22)
- [elasticsearch-pull-request #128411 / part-3](https://gradle-enterprise.elastic.co/s/mivdpc5yet36c)
- [elasticsearch-pull-request #127992 / part-3](https://gradle-enterprise.elastic.co/s/ec72tqndcppoa)
- [elasticsearch-pull-request #128101 / part-3](https://gradle-enterprise.elastic.co/s/ck2bd7kt6vbtw)
- [elasticsearch-pull-request #128095 / part-3](https://gradle-enterprise.elastic.co/s/vdklnwa7brbrc)
- [elasticsearch-periodic-platform-support #11966 / ubuntu-2404-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/iyzgyi2whhjv2)
- [elasticsearch-periodic-platform-support #11966 / ubuntu-2404-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/enf3kp2tlcfwk)
- [elasticsearch-pull-request #127992 / part-3](https://gradle-enterprise.elastic.co/s/6c3rzaa3iimas)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT.test" -Dtests.seed=76765A2A88A96B28 -Dtests.locale=el-Grek-GR -Dtests.timezone=US/Aleutian -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: query: from missing_ip_sample_data,sample__data_ts_nanos_lookup | grok message "%{WORD:JbelwLDJn}" | FORK  (where true) (where  NOT multi_match("data", client_ip, client_ip, message, {"type": "best_fields", "lenient": false})) (where event_duration == -96 AND event_duration < -25) | WHERE _fork == "fork3" | DROP _fork
exception: method [POST], host [http://[::1]:41013], URI [/_query?format=json&pretty=true&error_trace=true], status line [HTTP/1.1 400 Bad Request]
Warnings: [No limit defined, adding default limit of [1000]]
{
  "error" : {
    "root_cause" : [
      {
        "type" : "query_shard_exception",
        "reason" : "failed to create query: 'data' is not an IP string literal.",
        "index_uuid" : "EMdE0u66TLaXvggdD_P4qA",
        "index" : "sample__data_ts_nanos_lookup",
        "stack_trace" : "[sample__data_ts_nanos_lookup/EMdE0u66TLaXvggdD_P4qA] org.elasticsearch.index.query.QueryShardException: failed to create query: 'data' is not an IP str
[truncated]
```

**Issue Reasons:**
- [main] 20 failures in test test (4.4% fail rate in 454 executions)
- [main] 15 failures in step part-3 (6.0% fail rate in 249 executions)
- [main] 2 failures in step ubuntu-2404-aarch64_checkpart3_platform-support-arm (40.0% fail rate in 5 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-platform-support (40.0% fail rate in 5 executions)
- [main] 13 failures in pipeline elasticsearch-pull-request (5.4% fail rate in 240 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -402,15 +402,9 @@ tests:
 - class: org.elasticsearch.reindex.management.ReindexRelocationIT
   method: testLocalReindexRelocation
   issue: https://github.com/elastic/elasticsearch/issues/143643
-- class: org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT
-  method: test
-  issue: https://github.com/elastic/elasticsearch/issues/143023
 - class: org.elasticsearch.xpack.eql.action.EqlLoggingIT
   method: testEqlLoggingPartialShardFailure
   issue: https://github.com/elastic/elasticsearch/issues/143659
-- class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT
-  method: test
-  issue: https://github.com/elastic/elasticsearch/issues/143023
 - class: org.elasticsearch.xpack.security.support.MetadataFlattenedMigrationIntegTests
   method: testMigrationWithConcurrentUpdates
   issue: https://github.com/elastic/elasticsearch/issues/143674

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
@@ -70,6 +70,9 @@ public abstract class GenerativeRestTest extends ESRestTestCase implements Query
         "Inference endpoint not found \\[foo_inference_id\\]",
         // full-text functions are not allowed to match on fields that come from lookup indices
         "cannot operate on \\[.*\\], supplied by an index \\[.*\\] in non-STANDARD mode \\[lookup\\]",
+        "Can only use fuzzy queries on keyword and text fields - not on \\[.*\\] which is of type \\[.*\\]",
+        // multi_match query receiving a non-boolean value for a boolean type field
+        "Can't parse boolean value \\[.*\\], expected \\[true\\] or \\[false\\]",
 
         // Awaiting fixes for query failure
         "Unknown column \\[<all-fields-projected>\\]", // https://github.com/elastic/elasticsearch/issues/121741,
@@ -513,6 +516,7 @@ public abstract class GenerativeRestTest extends ESRestTestCase implements Query
      *   <li>Fields expanded by MV_EXPAND (the expanded fields)</li>
      *   <li>Fields created by GROK or DISSECT (the "extracted" fields)</li>
      *   <li>Fields renamed via RENAME</li>
+     *   <li>Fields produced by REGISTERED_DOMAIN (the sub-fields like prefix.domain, prefix.top_level_domain, etc.)</li>
      * </ul>
      * The error is allowed only when the offending field can be traced back to one of these commands.
      */
@@ -564,6 +568,11 @@ public abstract class GenerativeRestTest extends ESRestTestCase implements Query
                     if (EsqlQueryGenerator.unquote(rm.group(1).trim()).equals(fieldName)) {
                         return true;
                     }
+                }
+            } else if ("registered_domain".equals(name)) {
+                String prefix = (String) previous.context().get("prefix");
+                if (prefix != null && fieldName.startsWith(prefix + ".")) {
+                    return true;
                 }
             }
         }

